internal/core/src/index/VectorMemIndex.cpp

```diff
@@ -291,7 +295,7 @@ VectorMemIndex<T>::BuildWithDataset(const DatasetPtr& dataset,
     SetDim(dataset->GetDim());
 
     knowhere::TimeRecorder rc("BuildWithoutIds", 1);
-    auto stat = index_.Build(dataset, index_config);
+    auto stat = index_.Build(dataset, index_config, use_knowhere_build_pool_);
```

Comment:
This parameter get passed in this `VectorMemIndex` as class a field parameter but pass to Knowhere index class as a function parameter. Also, this `VectorMemIndex` to Knohwere index is 1:1 mapping. This sounds weird to me.

Reply:
Please see the comments in this place https://github.com/milvus-io/milvus/pull/41013/files#diff-5fd4c81a6e142b8c55a086a76a17c4ea5822ccbe825c3707fb33c1c657d81abeR31-R33
This is a quick fix. The most reasonable approach is to distinguish between writeable index and readonly index in segcore, because the interfaces of the two are actually different, which does not conform to the principle of single responsibility; this object will be refactored.